docs/DEVELOPMENT-GUIDE.md

```diff
@@ -347,10 +347,11 @@ The `high_water_mark` tracks the furthermost byte which has been written but not
 Note that this may be past the `write_cursor` if `s2n_stuffer_rewrite()` has been called.
 Explicitly tracking the `high_water_mark` allows us to track the bytes which need to be wiped, and helps avoids needless zeroing of memory.
 The next two bits of state track whether a stuffer was dynamically allocated (and so should be free'd later) and whether or not it is growable.
-`tainted` is set to 1 whenever the raw access functions are called.
-If a stuffer is currently tainted then it can not be resized and it becomes ungrowable.
+`tainted` is set to true whenever `s2n_stuffer_raw_read/write()` are called.
+`s2n_stuffer_raw_read/write()` returns an outstanding pointer that references a memory chunk within the stuffer. Whenever such an outstanding pointer exists, the stuffer's `tainted` flag must be set to true. However, if the outstanding pointer goes out of scope, the stuffer's `tainted` flag can safely be set to false.
```

Comment:
All this information is true, although I think it isn't phrased in a way that will really help someone understand what the 'tainted' flag means. This is why I recommend getting this reviewed by someone who is less familiar with this concept and seeing if they understand this explanation. 
```suggestion
`A 'tainted' stuffer therefore implies that an outstanding pointer currently exists from a previous `s2n_stuffer_raw_read/write()`call which points to memory within the stuffer. Of course, the tainted mark is not needed once the outstanding pointer goes out of scope. Therefore, the stuffer's tainted flag can be safely set to false when the outstanding pointer no longer exists, in a sense, the danger has passed.
```